### PR TITLE
Move safe_auth_ffi to a separate crate

### PR DESCRIPTION
**Pull request by ravinderjangra**
**Pending:**

- [x] Updating safe_auth_ffi dependency in all tests
- [x] safe_auth_ffi building issues
- [ ] CI

### CODE REVIEW

File: safe_app/Cargo.toml [hunk 1]
```diff
@@ -47,8 +49,8 @@ threshold_crypto = "~0.3.2"
 unwrap = "~1.2.0"
 
 [features]
-mock-network = ["testing", "safe_core/mock-network", "safe_authenticator/mock-network"]
-testing = ["safe_core/testing", "safe_authenticator/testing", "env_logger"]
+mock-network = ["testing", "safe_core/mock-network", "safe_authenticator/mock-network", "safe_authenticator_ffi/mock-network"]
+testing = ["safe_core/testing", "safe_authenticator/testing", "safe_authenticator/testing", "env_logger"]
```
Comment (lionel1704) [inline safe_app/Cargo.toml]:
```suggestion
testing = ["safe_core/testing", "safe_authenticator/testing", "safe_authenticator_ffi/testing", "env_logger"]
```

File: safe_authenticator/Cargo.toml [hunk 2]
```diff
@@ -35,15 +35,10 @@ safe_core = { path = "../safe_core", version = "~0.38.1", features = ["testing"]
...
 threshold_crypto = "~0.3.2"
 unwrap = "~1.2.0"
 
 [features]
 mock-network = ["testing", "safe_core/mock-network"]
 testing = ["safe_core/testing", "env_logger"]
-bindings = ["safe_bindgen"]
-
-[lib]
-crate_type = ["staticlib", "rlib", "cdylib"]
```
Comment (lionel1704) [inline safe_authenticator/Cargo.toml]: This is still needed. That's why CI was failing.

File: safe_authenticator_ffi/src/ffi/mod.rs [hunk 3]
```diff
@@ -122,11 +123,13 @@ pub unsafe extern "C" fn auth_reconnect(
...
 ) {
-    catch_unwind_cb(user_data, o_cb, || -> Result<_> {
+    catch_unwind_cb(user_data, o_cb, || -> AuthResult<_> {
         let user_data = OpaqueCtx(user_data);
         (*auth).send(move |client| {
             try_cb!(
-                client.restart_network().map_err(Error::from),
+                client
+                    .restart_network()
+                    .map_err(|_| Error::from("poda".to_string())),
```
Comment (lionel1704) [inline safe_authenticator_ffi/src/ffi/mod.rs]: :smile:

File: safe_authenticator_ffi/src/ffi/mod.rs [hunk 4]
```diff
@@ -127,9 +127,7 @@ pub unsafe extern "C" fn auth_reconnect(
         let user_data = OpaqueCtx(user_data);
         (*auth).send(move |client| {
             try_cb!(
-                client
-                    .restart_network()
-                    .map_err(|_| Error::from("poda".to_string())),
+                client.restart_network().map_err(|err| Error::from(err)),
```
Comment (lionel1704) [inline safe_authenticator_ffi/src/ffi/mod.rs]:
```suggestion
                client.restart_network().map_err(Error::from),
```
This will fix clippy

**Comment by ravinderjangra** [comment]
superseded by #1180

**Pull request closed without merging**